regression/cbmc-library/float-nan-check/main.c

```diff
@@ -0,0 +1,66 @@
+#include <assert.h>
+#include <math.h>
+#include <stdint.h>
+
+// This test contains a set of operations performed on a floating point numbers
+// to verify --nan-check flag. The result of test may change when/if support
+// of NaN value propagation checking is supported. Current behaviour of
+// --nan-check flag is that it checks whether an arithmetic operation performed
+// on a floating point value(s) is generating new not-a-number value. If NaN is
+// generated because one of the values is already NaN the assertion is not
+// produced.
+
+union container {
+  uint32_t u;
+  float f;
+};
+
+int main(void)
+{
+  union container c = {UINT32_MAX};
+  int n;
```

Comment:
`n` `int` or `float`?

Reply:
I don't think it really matters in my test, my thinking was that `n` as `int` should trigger implicit conversion to `float` on mixed type arithmetic, like `f = NAN + n;` but when I look at this from one day perspective it does not seem to prove anything.